pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DistinctFromTransformFunction.java

```diff
@@ -0,0 +1,124 @@
+ */
+public class DistinctFromTransformFunction extends BinaryOperatorTransformFunction {
+  // If _distinctType is 1, act as IsDistinctFrom transformation, otherwise act as IsNotDistinctFrom.
+  // This value should only be 0 or 1.
+  private final int _distinctType;
+
+  /**
+   * Returns a bit map of corresponding column.
+   * Returns null by default if null option is disabled.
+   */
+  @Nullable
+  private static RoaringBitmap getNullBitMap(ProjectionBlock projectionBlock, TransformFunction transformFunction) {
+    String columnName = ((IdentifierTransformFunction) transformFunction).getColumnName();
+    return projectionBlock.getBlockValueSet(columnName).getNullBitmap();
+  }
+
+  /**
+   * Returns true when bitmap is null (null option is disabled) or bitmap is empty.
+   */
+  private static boolean isEmpty(RoaringBitmap bitmap) {
+    return bitmap == null || bitmap.isEmpty();
+  }
+
+  /**
+   * @param distinctType is set to true for IsDistinctFrom, otherwise it is for IsNotDistinctFrom.
+   */
+  protected DistinctFromTransformFunction(boolean distinctType) {
+    super(distinctType ? TransformFunctionType.NOT_EQUALS : TransformFunctionType.EQUALS);
+    _distinctType = distinctType ? 1 : 0;
+  }
+
+  @Override
+  public String getName() {
+    if (_distinctType == 1) {
+      return TransformFunctionType.IS_DISTINCT_FROM.getName();
+    }
+    return TransformFunctionType.IS_NOT_DISTINCT_FROM.getName();
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    super.init(arguments, dataSourceMap);
+    if (!(_leftTransformFunction instanceof IdentifierTransformFunction)
+        || !(_rightTransformFunction instanceof IdentifierTransformFunction)) {
+      throw new IllegalArgumentException("Only column names are supported in DistinctFrom transformation.");
+    }
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return BOOLEAN_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    _results = super.transformToIntValuesSV(projectionBlock);
+    RoaringBitmap leftNull = getNullBitMap(projectionBlock, _leftTransformFunction);
+    RoaringBitmap rightNull = getNullBitMap(projectionBlock, _rightTransformFunction);
+    // Both sides are not null.
+    if (isEmpty(leftNull) && isEmpty(rightNull)) {
+      return _results;
+    }
+    // Left side is not null.
+    if (isEmpty(leftNull)) {
+      // Mark right null rows as distinct.
+      rightNull.forEach((IntConsumer) i -> _results[i] = _distinctType);
+      return _results;
+    }
+    // Right side is not null.
+    if (isEmpty(rightNull)) {
+      // Mark left null rows as distinct.
+      leftNull.forEach((IntConsumer) i -> _results[i] = _distinctType);
+      return _results;
+    }
+    RoaringBitmap xorNull = RoaringBitmap.xor(leftNull, rightNull);
```

Comment:
It should be faster if we iterate in the following way:
- Iterate over left null and set _distinctResult
- Iterate over right null and set _distinctResult
- Iterate over left AND right null and set _notDistinctResult

Reply:
Changed to iterate through xor and and bitset